critiquebrainz/frontend/views/artist.py

```diff
@@ -46,3 +48,50 @@ def entity(id):
         release_group['review_count'] = review_count
     return render_template('artist.html', id=id, artist=artist, release_type=release_type,
                            release_groups=release_groups, page=page, limit=limit, count=count)
+
+
+def squash_members_attributes(members):
+    members = [member.copy() for member in members]
+    ids_order = dict([(member['target'], index) for index, member in enumerate(members)])
+    
+    members.sort(key=lambda x: x['target'])
+    result_members = []
+
+    for i in range(0, len(members)):
+        if i == 0 or members[i]['target'] != members[i-1]['target']:
+            if not members[i].get('attribute-list', None):
+                members[i]['attribute-list'] = []
+            members[i]['periods'] = []
+            period = get_period(members[i])
+            if period:
+                members[i]['periods'].append(period)
+            j = i+1
+            while j < len(members) and members[j]['target'] == members[i]['target']:
+                members[i]['attribute-list'].extend(members[j].get('attribute-list', []))
+                period = get_period(members[i])
+                if(period):
+                    members[i]['periods'].append(period)
+                j += 1
+            result_members.append(members[i])
+            members[i]['periods'].sort(key=lambda x: x[1], reverse=True)
+
+    result_members.sort(key=lambda x: ids_order[x['target']])
+    return result_members
```

Comment:
This seems too complicated for what you are trying to do. What if you just build a list with a dictionary for each member:
```
{
  "name": ...,
  "periods": [..],
}
```